flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/ExpressionParser.scala

```diff
@@ -535,6 +536,13 @@ object ExpressionParser extends JavaTokenParsers with PackratParsers {
       case f ~ _ ~ _ => RowtimeAttribute(f)
     }
 
+  // key
+
+  lazy val key: PackratParser[Expression] =
+    (aliasMapping | "(" ~> aliasMapping <~ ")" | fieldReference) ~ "." ~ KEY ^^ {
```

Comment:
can we deduplicate `(aliasMapping | "(" ~> aliasMapping <~ ")" | fieldReference)`? I would expect so but I'm not familiar with this scala magic `ExpressionParser`.

Reply:
Do you mean deduplicate the code? I think we can add another `Expression`, e.g., `lazy val aliasOrFieldReference = aliasMapping | "(" ~> aliasMapping <~ ")" | fieldReference`, and reuse it.

Reply:
Yes, I meant that this `(aliasMapping | "(" ~> aliasMapping <~ ")" | fieldReference)` is duplicated couple of times here. If it is possible to deduplicate it to:
```
lazy val aliasOrFieldReference = aliasMapping | "(" ~> aliasMapping <~ ")" | fieldReference
```
I think it would be better to do so.